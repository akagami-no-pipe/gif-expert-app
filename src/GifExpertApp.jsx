
import {useState} from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {
    
    const [categories, setCategories] = useState([ 'One Punch', 'Dragon ball' ]);

    const onAddCategory = (category) => {
        if(categories.includes(category)) return;
        setCategories([category, ...categories]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onNewCategory={ (value) => onAddCategory(value) } />
                
            { 
                categories.map( (category) => (
                    <GifGrid key={category} category={category}/>
                ))
            }
        </>
    );
}
