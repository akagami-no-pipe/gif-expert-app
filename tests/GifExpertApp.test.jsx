import { render, screen, waitFor } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas en la app GifExpertApp', () => {

    test('Debe coincidir con el snapshot', () => {

        const { container } = render(<GifExpertApp/>);
        expect(container).toMatchSnapshot();

    });

    test('Debe venir el input vacio', () => {

        render(<GifExpertApp/>);
        const textbox = screen.getByRole('textbox');
        expect(textbox.value).toBeFalsy();

    });

    test('Debe venir el card-grid vacio', () => {

        const { container } = render(<GifExpertApp/>);
        const card_grid = container.getElementsByClassName('card-grid');
        expect(card_grid.length).toBe(1);
        expect(card_grid[0].hasChildNodes()).toBeFalsy();

    });

    test('Debe renderizar los gifs', async() => {

        const { container } = render(<GifExpertApp/>);
        const card_grid = container.getElementsByClassName('card-grid');
        await waitFor(
            () => expect(card_grid[0].childNodes.length).toBeGreaterThan(0)
        );
        expect(card_grid[0].hasChildNodes()).toBeTruthy();

    });

});