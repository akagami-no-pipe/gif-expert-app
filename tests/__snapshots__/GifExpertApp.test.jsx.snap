// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la app GifExpertApp Debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
  <h3>
    Full metal alchemist
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
